### PR TITLE
Update numpy 2.0.0

### PR DESCRIPTION
**Pull request by SDAllred**
This upgrades `numpy>2.0.0` so we can continue using it as a package in dquality.

To run tests locally, download the repo and run `make setup` and then `make test`. I found it easiest to do this inside a venv.

Changes:
- Ugrades Numpy
- Upgrades tensorflow from 2.16->2.19 (required by numpy)
- Updates the package name `numpy.lib.histograms` -> `numpy.lib._histograms_impl`(required by numpy)
- Updates to model behavior detailed below (required by tensorflow 2.19)

**Note:** The changes I make here are definitely a patch job. I think the best way to handle this would be to factor out a separate “training” model with only the softmax output, and point it at the same layers as the inference model. This would require significant effort on our part to understand and rearchitect their models and is probably not worth the effort. Also there is a ton of duplicate logic in here that I am choosing to leave alone.

In TF-Keras 2.19, any model with multiple outputs must be trained with a ground-truth `y_true` for every output. for example, their Character-Level CNN defines three outputs:

1. **Softmax** logits (`(batch, seq_len, num_labels)`) — the only one we actually train on  
2. **Argmax** indices (`(batch, seq_len)`) — downstream inference  
3. **Thresholded argmax** indices (`(batch, seq_len)`) — downstream inference  

We only care about the softmax for training, so:
- **Loss setup**  
  ```python
  self._model.compile(
      loss_weights={
        'softmax_output':   1.0,
        'argmax':           0.0, 
        'thresh_argmax':    0.0,
      },
      loss={
        'softmax_output': 'categorical_crossentropy',
        'argmax':         'mean_squared_error',
        'thresh_argmax':  'mean_squared_error',
      },
      …
  )
We assign a dummy loss (mean_squared_error) to the two inference outputs but zero their loss_weight, so they never affect gradients. Then at training time we supply dummy vectors as the `y` training vectors for the inference outputs to keep Keras happy.

Some tests rely on precompiled and saved versions of the models, and they have to be reconstructed in order for the tests to work after the changes I have made. I created the ability to optionally "force construct" models at load time to get around this. I would be better if these files were replaced, I don't feel super comfortable doing that without more information about how they were created.

There are 2 tests that still fail locally that I am choosing to ignore
1. testing that the seed is set to a default of `None`, but it's intentionally set to 0 when tests are run locally, so it fails.
2. The default verbosity of logs is `Warning` but should be `Info`, I think this is being set somewhere in testing and I'm not concerned about it.

### CODE REVIEW

File: dataprofiler/labelers/base_data_labeler.py [hunk 1]
```diff
@@ -535,7 +541,11 @@ def _load_model(
                 "and could not be found as a registered model "
                 "class in BaseModel.".format(str(model_class))
             )
-        self.set_model(model_class.load_from_disk(dirpath))
+
+        model = model_class.load_from_disk(dirpath)
+        if self.force_construct_model:
+            model._construct_model()
```
Comment (SDAllred) [inline dataprofiler/labelers/base_data_labeler.py:547]: There are saved models for testing in the repo that were compiled in tensorflow 2.16. After my model updates these will no longer work properly when making predictions. I am not super comfortable completely replacing these model files, but we can optionally force reconstruct the model after loading it using the update methods and successfully run the tests.

File: dataprofiler/labelers/char_load_tf_model.py [hunk 2]
```diff
@@ -262,7 +262,17 @@ def _construct_model(self) -> None:
 
         # Compile the model w/ metrics
         softmax_output_layer_name = self._model.output_names[0]
-        losses = {softmax_output_layer_name: "categorical_crossentropy"}
+        all_outputs = self._model.output_names
+
+        losses = {
+            softmax_output_layer_name: "categorical_crossentropy",
+            all_outputs[1]: "mean_squared_error",
```
Comment (SDAllred) [inline dataprofiler/labelers/char_load_tf_model.py:269]: this is a dummy loss that will be ignored

File: dataprofiler/labelers/char_load_tf_model.py [hunk 3]
```diff
@@ -262,7 +262,17 @@ def _construct_model(self) -> None:
...
+        }
+
+        loss_weights = {
+            softmax_output_layer_name: 1.0,
+            all_outputs[1]: 0.0,
```
Comment (SDAllred) [inline dataprofiler/labelers/char_load_tf_model.py:274]: setting the weights of the second loss to zero will cause the model to essentially ignore it, while still making it happy by passing in the same number of loss functions as outputs.

File: dataprofiler/labelers/char_load_tf_model.py [hunk 4]
```diff
@@ -381,19 +405,24 @@ def fit(
...
-        softmax_output_layer_name = self._model.output_names[0]
 
         start_time = time.time()
         batch_id = 0
         for x_train, y_train in train_data:
-            model_results = self._model.train_on_batch(
-                x_train, {softmax_output_layer_name: y_train}
-            )
+
+            dummy = np.zeros((y_train.shape[0], y_train.shape[1]), dtype=y_train.dtype)
```
Comment (SDAllred) [inline dataprofiler/labelers/char_load_tf_model.py:413]: Keras will error if any output is missing a corresponding `y` vector, since there are 2 outputs here but only the first one matters we use a dummy vector to satisfy Keras’s requirement without polluting the real training signal.

File: dataprofiler/tests/profilers/test_categorical_column_profile.py [hunk 5]
```diff
@@ -725,7 +725,7 @@ def test_categorical_diff(self):
                 "chi2-test": {
                     "chi2-statistic": 82 / 35,
                     "deg_of_free": 2,
-                    "p-value": 0.3099238764710244,
+                    "p-value": 0.3099238764710245,
```
Comment (SDAllred) [inline dataprofiler/tests/profilers/test_categorical_column_profile.py:728]: Very small change here, I feel ok just updating the tests.

File: dataprofiler/labelers/data_labelers.py [hunk 6]
```diff
@@ -126,7 +127,12 @@ def __new__(  # type: ignore
                 dirpath = os.path.join(
                     default_labeler_dir, data_labeler._default_model_loc
                 )
-            return TrainableDataLabeler(dirpath, load_options)
+            if force_construct_model:
```
Comment (kristiancailer) [inline dataprofiler/labelers/data_labelers.py:130]: Nitty but we can just pass `force_construct_model` and do `return TrainableDataLabeler(dirpath, load_options, force_construct_model)` in one line here.

File: dataprofiler/labelers/base_data_labeler.py (hunk 1 above)
Comment (kristiancailer) [reply dataprofiler/labelers/base_data_labeler.py:547]: I'm glad you found a way through this one by enabling model reconstruction!

**Review by kristiancailer — APPROVED** [review]

File: dataprofiler/labelers/data_labelers.py (hunk 6 above)
Comment (SDAllred) [reply dataprofiler/labelers/data_labelers.py:130]: So I tried this and tests started failing. Trying to figure out why was taking too long, so I am going to stick with it because it works.
